syncstorage/storage/sql/queries_generic.py

```diff
@@ -92,6 +93,47 @@
 DELETE_ITEMS = "DELETE FROM %(bso)s WHERE userid=:userid "\
                "AND collection=:collectionid AND id IN %(ids)s"
 
+CREATE_TRANSACTION = "INSERT INTO batch_uploads (batch, userid, collection) "\
+                     "VALUES (:batch, :userid, :collection)"
+
+VALID_TRANSACTION = "SELECT open FROM batch_uploads WHERE batch = :batch " \
+                    "AND userid = :userid AND collection = collection"
+
+APPEND_ITEMS_TO_TRANSACTION = "INSERT INTO batch_upload_items "\
+                              "  (batch, item, sortindex, modified, payload, "\
+                              "   payload_size, ttl) "\
+                              "VALUES "\
+                              "  (:batchid, :item, :sortindex, :modified, "\
+                              "   :payload, :payload_size, :ttl) "\
+                              "ON DUPLICATE KEY UPDATE "\
+                              "  sortindex = VALUES(sortindex), "\
+                              "  payload = VALUES(payload), "\
+                              "  payload_size = VALUES(payload_size) "\
+                              "  ttl = VALUES(ttl)"
+
+COMMIT_TRANSACTION = "INSERT INTO %(bso)s "\
+                     "  (userid, collection, id, sortindex, modified, "\
+                     "   payload, payload_size, ttl) "\
+                     "SELECT "\
+                     "  :userid, :collection, item, sortindex, modified, "\
+                     "  COALESCE(payload, \"\"), COALESCE(payload_size, 0), "\
+                     "  COALESCE(ttl, ttl + :default_ttl) "\
+                     "FROM %(batch_upload_items)s "\
+                     "WHERE batch = :batch "\
+                     "ON DUPLICATE KEY UPDATE "\
```

Comment:
Since `ON DUPLICATE KEY UPDATE` is a MySQL-specific extension, this version of the query should go into `queries_mysql` and a more generic one should be used here.  Is the sqlite version generic enough to work as the default implementation?